cpu/cc2538/periph/hwrng.c

```diff
@@ -63,25 +72,25 @@ void hwrng_init(void)
     }
 
     /* Seed the high byte first: */
-    SOC_ADC_RNDL = (seed >> 8) & 0xff;
-    SOC_ADC_RNDL = seed & 0xff;
+    soc_adc->RNDH = (seed >> 8) & 0xff;
+    soc_adc->RNDL = seed & 0xff;
 
     /* Turn RF off: */
     RFCORE_SFR_RFST = ISRFOFF;
 }
 
 void hwrng_read(void *buf, unsigned int num)
 {
-    unsigned count;
     uint8_t *b = (uint8_t *)buf;
 
-    for (count = 0; count < num; ) {
+    for (unsigned count = 0; count < num; count++) {
         /* Clock the RNG LSFR once: */
-        SOC_ADC->cc2538_adc_adccon1.ADCCON1bits.RCTRL = 1;
+        uint32_t reg32 = soc_adc->ADCCON1 | (1UL << SOC_ADC_ADCCON1_RCTRL_S);
+        soc_adc->ADCCON1 = reg32;
 
         /* Read up to 2 bytes of hwrng data: */
-        b[count++] = SOC_ADC_RNDL;
+        b[count++] = soc_adc->RNDL;
```

Comment:
You cleaned up the other inline incrementation. Putting this increment explicit on a following line would be perfect.

Reply:
done, I also adapted the logic of the `if` statement and to match coding style.

plus also inlined the read-write of the register as discussed offline, in a separate commit